rust/perspective/arrow_accessor/tests/util.rs

```diff
@@ -0,0 +1,332 @@
+            NaiveDateTime::parse_from_str("2020-01-15 12:30:45", "%Y-%m-%d %H:%M:%S")
+                .unwrap()
+                .timestamp(),
+            NaiveDateTime::parse_from_str("2020-02-29 22:55:45", "%Y-%m-%d %H:%M:%S")
+                .unwrap()
+                .timestamp(),
+            NaiveDateTime::parse_from_str("2020-03-30 05:12:45", "%Y-%m-%d %H:%M:%S")
+                .unwrap()
+                .timestamp(),
+            NaiveDateTime::parse_from_str("2020-04-01 09:59:45", "%Y-%m-%d %H:%M:%S")
+                .unwrap()
+                .timestamp(),
+            NaiveDateTime::parse_from_str("2020-05-15 19:01:45", "%Y-%m-%d %H:%M:%S")
+                .unwrap()
+                .timestamp(),
+        ])),
+        Arc::new(BooleanArray::from(vec![true, false, true, false, true])),
+    ];
+
+    if with_string_column {
+        let strings = vec!["abc", "def", "abc", "hij", "klm"];
+        let dict_array: DictionaryArray<Int32Type> = strings.into_iter().collect();
+        arrays.push(Arc::new(dict_array));
+    }
+
+    // Arrow schema created using `Field`
+    let mut schema = Schema::new(vec![
+        Field::new("a", DataType::UInt8, false),
+        Field::new("b", DataType::UInt16, false),
+        Field::new("c", DataType::UInt32, false),
+        Field::new("d", DataType::UInt64, false),
+        Field::new("e", DataType::Int8, false),
+        Field::new("f", DataType::Int16, false),
+        Field::new("g", DataType::Int32, false),
+        Field::new("h", DataType::Int64, false),
+        Field::new("i", DataType::Float32, false),
+        Field::new("j", DataType::Float64, false),
+        Field::new("k", DataType::Date32(DateUnit::Day), false),
+        Field::new("l", DataType::Timestamp(TimeUnit::Millisecond, None), false),
+        Field::new("m", DataType::Boolean, false),
+    ]);
+
+    if with_string_column {
+        let dict_field = Field::new(
+            "n",
+            DataType::Dictionary(Box::new(DataType::Int32), Box::new(DataType::Utf8)),
+            false,
+        );
+        schema = Schema::try_merge(&vec!(schema, Schema::new(vec![dict_field]))).unwrap();
+    }
+
+    let schema_ref = SchemaRef::new(schema);
+
+    let try_batch = RecordBatch::try_new(schema_ref.clone(), arrays);
+
+    // Accessor schema of names to `DataType`
+    let mut accessor_schema: HashMap<String, DataType> = HashMap::new();
+
+    for field in schema_ref.fields() {
+        accessor_schema.insert(field.name().clone(), field.data_type().clone());
+    }
+
+    match try_batch {
+        Ok(batch) => Box::new(batch),
+        Err(err) => panic!(err)
+    }
+}
+
+// Generate an Arrow record batch with null values in columns, only including
+// a string column if with_string_column is true as the StreamWriter doesn't
+// seem to write dictionary arrays properly - they cannot be parsed in
+// Rust or in pyarrow at the moment.
+pub fn make_arrow_nullable(with_string_column: bool) -> Box<RecordBatch> {
```

Comment:
https://github.com/apache/arrow/commit/5ed4695c4d08a60d1dec451182bde33d14b00161 might address this issue. IF you can, may you please check with the latest HEAD. Otherwise this is an issue that we should address before the next Arrow release.

Reply:
Will check with latest HEAD - I'll get a minimal pure Rust repro if that doesn't work out.

Reply:
Fixed in latest Git master - thank you for the heads up!